File: bro-annotations/src/main/java/me/xx2bab/bro/annotations/BroActivity.java
package me.xx2bab.bro.annotations;

import java.lang.annotation.Documented;
import java.lang.annotation.ElementType;
import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;
import java.lang.annotation.Target;

import me.xx2bab.bro.common.IBroModule;


@Documented
@Target(ElementType.TYPE)
@Retention(RetentionPolicy.RUNTIME)
public @interface BroActivity {

    String alias();

    Class<? extends IBroModule> module();

}

File: bro-annotations/src/main/kotlin/me/xx2bab/bro/annotations/BroActivity.kt
package me.xx2bab.bro.annotations

import me.xx2bab.bro.common.IBroModule
import kotlin.reflect.KClass

@MustBeDocumented
@Target(AnnotationTarget.ANNOTATION_CLASS, AnnotationTarget.CLASS)
@kotlin.annotation.Retention(AnnotationRetention.RUNTIME)
annotation class BroActivity(val alias: String, val module: KClass<out IBroModule>)